### ISSUE TITLE
Final Peer Review - rl447

### ISSUE THREAD

**Issue by rohanlewis** [GaryLeeONE/BaseballSTW#16]
### Summary:
This project aims to apply data analysis models and techniques to 2019 play-by-play MLB data to predict whether a hitter in Major League Baseball will swing and make contact, not swing, or swing and miss at a certain pitch based on the batter’s hitting tendencies, the pitch’s characteristics, and the game situation. This can greatly benefit pitchers and provide insight into suggested pitches to use against certain hitters and other key factors that can influence batting outcomes. The dataset contains 743,572 example plays and 90 features and was scraped from the Baseball Savant website using the Statcast search feature. 

### Things I Like:
1. I appreciated that you had an extra note at the bottom with an explanation for readers not familiar with baseball. Coming from a similar position where our project was predicting the point differential of NFL games, we felt that in the final report it was crucial to provide more background information to the audience and assume they don’t know about the topic, and having this information optionally available in the bottom of the report is a thoughtful addition.
2. The data visualizations you included were not only visually appealing but extremely informative in the exploratory analysis section. I was most impressed by the heat map showing the relationship between pitch location and the swing+whiff rate. I think that seeing this visualization leads nicely into the radial kernel choice for your SVM, and without this visualization that might not have been so obvious.
3. In the initial processing and aggregating of the dataset sections, it is clear that you took several important factors into account to filter your dataset. One issue that I saw with play by play data in our own project was the massive amounts of noise with so many features, and I appreciated that you intelligently selected them and only included hitters that have seen more than 1,000 pitches during the 2019 season while managing to maintain a robust dataset of 583,581 pitches with 41 features.

### Areas for Improvement:
1. I see that you plotted the variable importance metric from the XGBoost model. It would be interesting to see how this graph may differ across models, such as the EBM model provided by the InterpretML package. This could help inform further feature selection and engineering.
2. I thought the confusion matrix you provided showing the prediction statistics by class for the two models was very interesting, but I would not have expected that the XGBoost model can better predict both true positive swings and true negative non-swings. One idea to consider is that the model performs only marginally better on whiffs because it is potentially overfitting on swings and takes.
3. I am interested to see how psychology, past performance, and potential behavioral economic trends such as the “hot hand” fallacy might play a role in the prediction of a swing. In your initial processing step you mentioned that you removed the feature that described each play, and another potential area to explore would be how a batter’s recent past success or lack thereof might influence the batter’s decisions. This may require some natural language processing to extract the outcome of recent pitches from the text column.
